include/domain-types.h

```diff
@@ -16,14 +17,73 @@ namespace connect
 
 typedef int32_t id_t;
 
-typedef int status_t;
-
-enum
+class Status
 {
-    STATUS_OK       = 0,
-    STATUS_ERROR    = 1
+public:
+    Status(int code, bool isError, int facility)
+    {
+        assert(0 <= code  &&  code <= MAX_CODE);
+        assert(0 <= facility  &&  facility <= MAX_FACILITY);
```

Comment:
At first I made them uint16_t, then changed to int. Reason: uint16_t hardly save anything, but may cause (implicit) conversions.